File: Caesar_Cipher/CrackCaesarBruteForce.java
package Caesar_Cipher;

public class CrackCaesarBruteForce {
    private static final String ALPHABET = " ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public static void bruteForceCrack(String cipherText){
        for( int key=0; key<ALPHABET.length(); key++){
            StringBuilder plainText = new StringBuilder();
            for(char ch: cipherText.toCharArray()){
                int index = Math.floorMod(ALPHABET.indexOf(ch)-key, ALPHABET.length());
                plainText.append(ALPHABET.charAt(index));
            }
            System.out.println("for key: "+key+", plainText: "+plainText.toString());
        }
    }
    public static void bruteForceCrackWithLangDetection(String cipherText){
        // System.out.println("here");
        for( int key=0; key<ALPHABET.length(); key++){
            StringBuilder plainText = new StringBuilder();
            for(char ch: cipherText.toCharArray()){
                int index = Math.floorMod(ALPHABET.indexOf(ch)-key, ALPHABET.length());
                plainText.append(ALPHABET.charAt(index));
            }
            
            if(LanguageDetection.isEnglishLanguage(plainText.toString(), 70)){
                System.out.format("Caesar cipher cracked, key =%d, Plain Text is: %s%n", key, plainText.toString());
            }
        }
    }
    public static void main(String[] args) {
        String cipherText = CaesarCipher.encrypt("This is message", 3);
        LanguageDetection.loadData();
        bruteForceCrackWithLangDetection(cipherText);
    }
}

File: Caesar_Cipher/crack_caesar_brute_force.py

import caesar_cipher
import language_detector

ALPHABET = caesar_cipher.ALPHABET

def brute_force_crack(cipher_text):
    for key in range(len(ALPHABET)):
        plain_text = ''
        for c in cipher_text:
            index = ALPHABET.find(c)
            index = (index-key) % len(ALPHABET)
            plain_text = plain_text + ALPHABET[index]
        # print(language_detector.is_english_text(plain_text, 90))
        if language_detector.is_english_text(plain_text, 70):
            print('We have cracked Caesar Cipher, key is:', key, ', Plain_text is:', plain_text)
        # print('for key:', key, ', plain_text:', plain_text)

if __name__ == '__main__':
    language_detector.get_data()
    cipher_text = caesar_cipher.encrypt('This is message I am awkward', 3)
    brute_force_crack(cipher_text=cipher_text)